src/Chapter2.hs

```diff
@@ -857,7 +896,12 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind list =
+  if null list
+    then []
+    else list !! lastIndex : rewind (take lastIndex list)
+  where lastIndex = length list - 1
```

Comment:
Your solution is correct! However, it is slow. In lists, it is quite slow take anything from the end of the list. ALso you would use `length` at each iteration which is also very expensive. 
That is why a more efficient solution is with the accumulator and the recursive function that will do the addition at the start of the list which is instant!

You can read a bit more about the `go` pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go